src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,16 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.Warn("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Message types used for publishing should not implement interfaces.");
+
+                if (interfaces.Count > 1)
+                {
+                    Logger.ErrorFormat("Publishing type has multiple interfaces. Interfaces are {0}, the one that will be used for building routing key is {1}", string.Join(", ", interfaces), implementedInterface);
```

Comment:
@adamralph I think the idea here is that if we detect any interfaces at all, we log the warning, but when we find more than one, it should be logged as an error, because that is definitely going to break. If an endpoint subscribes to one of the interfaces that isn't chosen here, it will not receive any messages.